app/models/accounting/interest_calculator.rb

```diff
@@ -77,81 +77,89 @@ def recalculate
         # transactions, except the date of the first transaction.
         txns = []
 
+        # this adds an interest transaction
+        # int txns should always be managed (for now)
         txns << loan.transactions.build(
           txn_date: date,
           amount: 0, # Will be updated momentarily.
           loan_transaction_type_value: Transaction::LOAN_INTEREST_TYPE,
-          description: I18n.t('transactions.interest_description', loan_id: loan.id)
+          description: I18n.t('transactions.interest_description', loan_id: loan.id),
+          managed: true
         ) if add_int_tx?(txns_by_date[date], prev_tx)
 
         txns.concat(txns_by_date[date]) if txns_by_date[date]
 
         txns.each do |tx|
-          case tx.loan_transaction_type_value
-            when "interest"
-              tx.amount = accrued_interest(prev_tx, tx)
-              line_item_for(tx, int_rcv_acct).assign_attributes(
-                qb_line_id: 0,
-                posting_type: "Debit",
-                amount: tx.amount
-              )
-              line_item_for(tx, int_inc_acct).assign_attributes(
-                qb_line_id: 1,
-                posting_type: "Credit",
-                amount: tx.amount
-              )
-
-            when "disbursement"
-              line_item_for(tx, prin_acct).assign_attributes(
-                qb_line_id: 0,
-                posting_type: "Debit",
-                amount: tx.amount
-              )
-              line_item_for(tx, tx.account).assign_attributes(
-                qb_line_id: 1,
-                posting_type: "Credit",
-                amount: tx.amount
-              )
-
-            when "repayment"
-              int_part = [tx.amount, prev_tx.try(:interest_balance) || 0].min
-              line_item_for(tx, tx.account).assign_attributes(
-                qb_line_id: 0,
-                posting_type: "Debit",
-                amount: tx.amount
-              )
-              line_item_for(tx, prin_acct).assign_attributes(
-                qb_line_id: 1,
-                posting_type: "Credit",
-                amount: tx.amount - int_part
-              )
-              line_item_for(tx, int_rcv_acct).assign_attributes(
-                qb_line_id: 2,
-                posting_type: "Credit",
-                amount: int_part
-              )
+          if tx.managed?
```

Comment:
yea...so that was actually the base of the story

`next` would skip out the entire loop and this didn't allow us to recalculate balances so it kept returning 0